apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/DbUtils.java

```diff
@@ -171,6 +171,17 @@ public static String limitsClause(int from, int to) {
         }
     }
 
+    public static String limitsClauseInShardDb(int from, int to) {
+        int limit = to >=0 && to >= from && to < Integer.MAX_VALUE ? to - from + 1 : 0;
+        if (limit > 0) { // offset doesn't work in shard db because records were moved
```

Comment:
Why offset doesnt work?

Reply:
Because OFFSET is valid only for full database that was not split to shards.
Split database doesn't have previous offset anymore.
Let's we have two cases :
1.
10 rows in full database. Offset = 5 points to range row numbers = 5...10
2.
We've created two shards like
5 rows in shard  database #1.
and 5 rows in shard  database #2.
Now **offset = 5** doesn't make sense in shard #1 because offset =5 out of db #1 range.

**offset = 5** doesn't make sense in shard #2 too, because offset = 5 points to end of the shard #2. But correct data is within offset = 0...5 for shard #2

https://www.essentialsql.com/what-is-the-differenence-between-top-and-offset-fetch/
**....OFFSET and FETCH are used to return a window of records from a result set.  OFFSET specifies how many rows to skip within the result, and FETCH specifies how many rows from that point forward to return in the result.....**